pkg/binlog-filter/filter.go

```diff
@@ -115,14 +124,42 @@ func (b *BinlogEventRule) Valid() error {
 		return errors.Errorf("action of binlog event rule %+v should not be empty", b)
 	}
 
-	// TODO: check validity of dml/ddl event.
 	for i := range b.Events {
-		b.Events[i] = EventType(strings.ToLower(string(b.Events[i])))
+		et, err := b.toEvent(string(b.Events[i]))
+		if err != nil {
+			return errors.Annotatef(ErrInvalidEventType, "event type %s", b.Events[i])
+		}
+		b.Events[i] = et
 	}
-
 	return nil
 }
 
+// ErrInvalidEventType is returned when event type is not valid.
+var ErrInvalidEventType = errors.New("event type not found")
+
+// toEvent converts event type string to EventType and check if it is valid.
+func (b *BinlogEventRule) toEvent(es string) (EventType, error) {
```

Comment:
why this function is a method of `BinlogEventRule`? maybe you want iterate through b.Events inside this function?

Reply:
good catch, it should not be a method of `BinlogEventRule`.